pkg/model/deployment.go

```diff
@@ -155,13 +152,14 @@ func (b *BackstageDeployment) setReplicas(replicas *int32) {
 // sets container image name of Backstage Container
 func (b *BackstageDeployment) setImage(image *string) {
 	if image != nil {
-		// TODO this is a workaround for RHDH/Janus configuration
+		b.container().Image = *image
+		// this is a workaround for RHDH/Janus configuration
 		// it is not a fact that all the containers should be updated
-		// in general case need something smarter (probably annotation based)
+		// in general case need something smarter
 		// to mark/recognize containers for update
-		VisitContainers(b.podSpec(), func(container *corev1.Container) {
-			container.Image = *image
-		})
+		if len(b.podSpec().InitContainers) > 0 {
+			b.podSpec().InitContainers[0].Image = *image
```

Comment:
Why does this assume that the first init container is the one to modify? Same in `b.container()` by the way.. I guess this might cause some issues if people prepend additional init containers or containers to the list.
In the fix for 1.1.x (https://github.com/janus-idp/operator/pull/279/files#diff-8e7e7c9a64394f1ee039417ef32b84b7122580096d39f293a8d976bd2af27f45R237), we were leveraging the container names, which I think is a safer alternative, no?

Reply:
both equally ugly :) but ok for now, changed